### ISSUE TITLE
Incompatible with some mod?

### ISSUE THREAD

**Issue by HySand** [MehVahdJukaar/sleep_tight#18]
<details><summary>MOD list</summary>
<p>

Loading 274 mods:
	- adorn 3.8.1+1.19.2
	- advanced_runtime_resource_pack 0.6.2 via enhancedblockentities
	- advancementplaques 1.4.6
	- ambientsounds 5.2.13
	- amecsapi 1.3.7+mc22w17a via mousewheelie
	- appleskin 2.4.1+mc1.19
	- architectury 6.5.69
	- backslot 1.2.11
	- balm-fabric 4.5.2+0
	- bclib 2.1.6
	- betteradvancements 0.3.0.148
	- betteranimationscollection 4.0.5
	- bettercombat 1.5.4+1.19
	- betterend 2.1.4
	- bettermounthud 1.2.0
	- betternether 7.1.3
	- betterpingdisplay 1.1.1
	- blue_endless_jankson 1.2.1 via owo
	- blur 2.6.0
	- bobby 4.0.1
	- bookshelf 16.2.16
	- caffeineconfig 1.0.0+1.17 via sodium-extra
	- cakechomps 6.0.0+1.19.2
	- cameraoverhaul 1.4.0-fabric-universal
	- cardinal-components-base 5.0.2 via numismatic-overhaul
	- cardinal-components-block 5.0.2 via puzzleslib
	- cardinal-components-chunk 5.0.2 via puzzleslib
	- cardinal-components-entity 5.0.2 via numismatic-overhaul
	- cardinal-components-world 5.0.2 via puzzleslib
	- carpeted 1.19.2-1.4
	- cavedust 1.2.0
	- chat-up 2.1
	- chat_heads 0.9.0
	- chatsounds 1.1.7
	- chunksfadein 3.0.9-1.19
	- cloth-api 4.0.65
	- cloth-basic-math 0.6.1 via cloth-api
	- cloth-client-events-v0 4.0.65 via cloth-api
	- cloth-common-events-v1 4.0.65 via cloth-api
	- cloth-config 8.2.88
	- cloth-scissors-api-v1 4.0.65 via cloth-api
	- cloth-utils-v1 4.0.65 via cloth-api
	- coat 1.0.0-beta.19+mc22w17a via mousewheelie
	- com_electronwill_night-config_core 3.6.5 via couplings
	- com_electronwill_night-config_toml 3.6.5 via couplings
	- com_github_llamalad7_mixinextras 0.1.1 via porting_lib
	- com_google_code_findbugs_jsr305 3.0.2 via create
	- com_logisticscraft_occlusionculling 0.0.6-SNAPSHOT via entityculling
	- com_moandjiezana_toml_toml4j 0.7.2 via chunksfadein
	- com_typesafe_config 1.4.2 via bobby
	- combatroll 1.1.3+1.19
	- conditional-mixin 0.3.0 via moreculling
	- continuity 2.0.1+1.19
	- cosmetic-armor 1.4.0
	- couplings 1.9.1+1.19
	- create 0.5.0.i-961+1.19.2
	- creativecore 2.9.3
	- cursormod 1.3.4
	- damagetilt 1.19-fabric-0.1.2
	- debugify 2.8.0
	- diagonalfences 4.2.3
	- dramaticdoors 1.19.2-1.4.2_1
	- dripstone_fluid_lib 2.0.1 via milk
	- droplight 1.0.5
	- easyanvils 4.0.6
	- easymagic 4.3.2
	- effective 1.4
	- enchant_giver 1.3.0 via mcdw
	- enchdesc 13.0.14
	- enhancedblockentities 0.7.2+1.19.2
	- entityculling 1.5.2-mc1.19
	- error_notifier 1.0.9 via roughlyenoughitems
	- exordium 1.0.1-mc1.19.2
	- extended_armor_bars 1.2.1
	- fabric-api 0.75.1+1.19.2
	- fabric-api-base 0.4.15+8f4e8eb390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.14+93d8cb8290 via fabric-api
	- fabric-biome-api-v1 9.1.0+82db024a90 via fabric-api
	- fabric-block-api-v1 1.0.2+e415d50e90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.25+cafc6e8e90 via fabric-api
	- fabric-client-tags-api-v1 1.0.5+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.16+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.2.1+413cbbc790 via fabric-api
	- fabric-commands-v0 0.2.33+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.41+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.5.1+7c6cd14d90 via fabric-api
	- fabric-convention-tags-v1 1.2.2+27265b3490 via fabric-api
	- fabric-crash-report-info-v1 0.2.8+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.3.7+413cbbc790 via fabric-api
	- fabric-diagonal-panes 0.6.4
	- fabric-dimensions-v1 2.1.35+0d0f210290 via fabric-api
	- fabric-entity-events-v1 1.5.4+9244241690 via fabric-api
	- fabric-events-interaction-v0 0.4.34+562bff6e90 via fabric-api
	- fabric-events-lifecycle-v0 0.2.35+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.24+b6b6abb490 via fabric-api
	- fabric-item-api-v1 1.6.5+dce4603e90 via fabric-api
	- fabric-item-groups-v0 0.3.38+9244241690 via fabric-api
	- fabric-key-binding-api-v1 1.0.25+5c4fce2890 via fabric-api
	- fabric-keybindings-v0 0.2.23+df3654b390 via fabric-api
	- fabric-language-kotlin 1.9.1+kotlin.1.8.10
	- fabric-lifecycle-events-v1 2.2.4+1b46dc7890 via fabric-api
	- fabric-loot-api-v2 1.1.12+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.15+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.7+93d8cb8290 via fabric-api
	- fabric-mining-level-api-v1 2.1.23+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.21+c6af733c90 via fabric-api
	- fabric-networking-api-v1 1.2.11+10eb22f490 via fabric-api
	- fabric-networking-v0 0.3.28+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.2.1+d8ef690890 via fabric-api
	- fabric-particles-v1 1.0.14+4d0d570390 via fabric-api
	- fabric-recipe-api-v1 1.0.1+413cbbc790 via fabric-api
	- fabric-registry-sync-v0 0.9.32+9244241690 via fabric-api
	- fabric-renderer-api-v1 1.2.1+1adbf27790 via fabric-api
	- fabric-renderer-indigo 0.8.0+1adbf27790 via fabric-api
	- fabric-renderer-registries-v1 3.2.24+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.19+6e0787e690 via fabric-api
	- fabric-rendering-fluids-v1 3.0.11+4d0d570390 via fabric-api
	- fabric-rendering-v0 1.1.27+df3654b390 via fabric-api
	- fabric-rendering-v1 1.12.1+d8ef690890 via fabric-api
	- fabric-resource-conditions-api-v1 2.1.2+aae9039d90 via fabric-api
	- fabric-resource-loader-v0 0.8.3+251d254a90 via fabric-api
	- fabric-screen-api-v1 1.0.32+4d0d570390 via fabric-api
	- fabric-screen-handler-api-v1 1.3.7+1cc24b1b90 via fabric-api
	- fabric-sound-api-v1 1.0.2+c4f28df590 via fabric-api
	- fabric-textures-v0 1.0.24+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.1.6+413cbbc790 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.3+08b73de490 via fabric-api
	- fabricloader 0.14.12
	- fallingleaves 1.13.0+1.19.2
	- fancymenu 2.13.2
	- farmersdelight 1.19.2-1.3.9
	- fastload 2.6.11
	- ferritecore 5.0.3
	- fiber 0.23.0-2 via healthoverlay
	- fishofthieves 2.2.1
	- flywheel 0.6.8.a-4 via create
	- forge_tags 3.0 via porting_lib
	- forgeconfigapiport 4.2.11
	- ftbchunks 1902.3.15-build.233
	- ftblibrary 1902.3.15-build.188
	- ftbteams 1902.2.12-build.95
	- fwaystones 3.0.7+mc1.19.2
	- healthoverlay 7.2.3
	- held-item-info 1.4.0
	- highlighter 1.1.4
	- hitmarker 1.0a-1.16.2
	- iceberg 1.0.46
	- indium 1.0.9+mc1.19.2
	- ingameime 1.6.6
	- io_github_douira_glsl-transformer 2.0.0-pre8 via iris
	- io_leangen_geantyref_geantyref 1.3.13 via bobby
	- iris 1.5.0
	- itemborders 1.1.6
	- itemmodelfix 1.0.3+1.19
	- jade 8.7.0
	- jadeaddons 3.0.0
	- java 17
	- kanos_config 0.3.1+1.14.4-1.19 via statement
	- kirin 1.11.1 via presencefootsteps
	- konkrete 1.6.0
	- lambdabettergrass 1.3.0+1.19
	- lambdynlights 2.2.0+1.19.2
	- lazydfu 0.1.3
	- legendarytooltips 1.3.3
	- lootr 0.4.25.64
	- mcda 2.2.0
	- mcdw 5.0.4-1.19
	- mcwdoors 1.0.8
	- mcwfences 1.0.7
	- mcwlights 1.0.4
	- mcwpaintings 1.0.4
	- mcwwindows 2.1.2
	- memoryleakfix 1.19.3-0.7.0
	- midnightlib 0.5.2 via blur
	- milk 1.0.51 via create
	- minecraft 1.19.2
	- mm 2.3 via porting_lib
	- modmenu 4.1.2
	- moonlight 1.19.2-2.2.22
	- morechathistory 1.1.1
	- moreculling 1.19.1-0.15.0
	- morevillagers 4.0.2
	- mousewheelie 1.10.7+mc1.19.2
	- nimble 3.0.0
	- notenoughanimations 1.6.2
	- numismatic-overhaul 0.2.8+1.19
	- org_anarres_jcpp 1.4.14 via iris
	- org_antlr_antlr4-runtime 4.11.1 via iris
	- org_javassist_javassist 3.28.0-GA via yungsapi
	- org_jetbrains_kotlin_kotlin-reflect 1.8.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.8.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.8.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.8.10 via fabric-language-kotlin
	- org_jetbrains_kotlinx_atomicfu-jvm 0.19.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.4.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.4.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.4.1 via fabric-language-kotlin
	- org_joml_joml 1.10.4 via sodium
	- org_reflections_reflections 0.10.2 via yungsapi
	- org_spongepowered_configurate-core 4.1.2 via bobby
	- org_spongepowered_configurate-hocon 4.1.2 via bobby
	- owo 0.9.3+1.19
	- patchouli 1.19.2-77-FABRIC
	- pickupnotifier 4.2.0
	- player-animator 0.4.0
	- porting_lib 2.1.784+1.19.2 via create
	- porting_lib_accessors 2.1.784+1.19.2 via porting_lib
	- porting_lib_attributes 2.1.784+1.19.2 via porting_lib
	- porting_lib_base 2.1.784+1.19.2 via porting_lib
	- porting_lib_common 2.1.784+1.19.2 via porting_lib
	- porting_lib_constants 2.1.784+1.19.2 via porting_lib
	- porting_lib_entity 2.1.784+1.19.2 via porting_lib
	- porting_lib_extensions 2.1.784+1.19.2 via porting_lib
	- porting_lib_fake_players 2.1.784+1.19.2 via porting_lib
	- porting_lib_lazy_registration 2.1.784+1.19.2 via porting_lib
	- porting_lib_model_generators 2.1.784+1.19.2 via porting_lib
	- porting_lib_model_loader 2.1.784+1.19.2 via porting_lib
	- porting_lib_models 2.1.784+1.19.2 via porting_lib
	- porting_lib_networking 2.1.784+1.19.2 via porting_lib
	- porting_lib_obj_loader 2.1.784+1.19.2 via porting_lib
	- porting_lib_transfer 2.1.784+1.19.2 via porting_lib
	- presencefootsteps 1.6.4
	- pride 1.1.2+1.19 via lambdynlights
	- prism 1.0.3
	- puzzleslib 4.3.12
	- raknetify 0.1.0+alpha.5.61
	- reach-entity-attributes 2.3.0 via create
	- reeses-sodium-options 1.4.9+mc1.19.2-build.67
	- registrate-fabric 1.1.50-MC1.19.2 via create
	- rottencreatures 1.0.1
	- roughlyenoughitems 9.1.591
	- satin 1.9.0
	- serialization_hooks 0.3.26 via porting_lib
	- skinlayers 1.5.2-mc1.19.1
	- sleep_tight 1.19.2-1.0.12
	- smoothswapping 0.6
	- sodium 0.4.4+build.18
	- sodium-extra 0.4.16+mc1.19.2-build.90
	- sound_physics_remastered 1.19.2-1.0.18
	- spiceoffabric 1.6.0-beta.2+1.19.2
	- spruceui 4.1.0+1.19.2 via lambdynlights
	- stacc 1.3.3 via numismatic-overhaul
	- starlight 1.1.1+fabric.ae22326
	- statement 4.2.4+1.14.4-1.19.1
	- statement_vanilla_compatibility 1.0.1+1.16.5-1.17 via statement
	- status-effect-bars 1.0.1
	- supplementaries 1.19.2-2.2.60
	- talkbubbles 1.0.4
	- tectonic 1.1.3
	- tia 1.19.2-1.0.0
	- tinyskeletons 4.2.2
	- trashslot 12.0.1
	- travelerstitles 1.19.2-Fabric-3.1.0
	- treechop 0.17.6
	- trinkets 3.4.1
	- tweed4_annotated 1.3.1+mc22w17a via mousewheelie
	- tweed4_base 1.7.1+mc22w17a via mousewheelie
	- tweed4_data 1.2.1+mc22w17a via mousewheelie
	- tweed4_data_hjson 1.1.1+mc22w17a via mousewheelie
	- tweed4_tailor_coat 1.1.3+mc22w17a via mousewheelie
	- tweed4_tailor_screen 1.1.3+mc22w17a via mousewheelie
	- visuality 0.5.5
	- visualworkbench 4.2.1
	- vmp 0.2.0+beta.7.27
	- waterdripsound 1.19-0.3.0
	- waveycapes 1.2.2
	- waystones 11.1.0
	- xlpackets 1.19.2-4
	- yet-another-config-lib 1.7.1
	- yigd 1.4.8
	- yungsapi 1.19.2-Fabric-3.8.6

</p>
</details>

  The title bar shows `Press LShift to unmount` instead of `Press space to sleep.....` after right clicking a bed. I tried to disable some mods like Adorn \ Create \ better mount HUD \ Camera overhaul \ Exordium \ Advancement Plaques \ Travelers Titles \ Combat Roll \ Chat up but failed. 

  Maybe I messed up some settings. I'm going to test further to find the problem later.

**Comment by MehVahdJukaar** [comment]
Hm some mod must be having a mixin in same place I have. Can you try removing some until it works again?

**Comment by HySand** [comment]
May be caused by [VMP](https://github.com/RelativityMC/VMP-fabric)

**Comment by BLUEChester** [comment]
Yeah just did some testing and the mod is certainly incompatible with Very Many Players, attacking any entity (villager, armour stand, etc) will cause the game to completely crash. Not sure if the original issue in this post is related.

**Comment by MehVahdJukaar** [comment]
i just tried booting up with that mod and attacking some entities in single player and didnt get any crash. send your log

**Comment by BLUEChester** [comment]
After further testing, It doesn't actually seem to be caused by Very Many Players, sorry for that. It is, however, caused by '[Sleep Warp](https://modrinth.com/mod/sleep-warp)'. 

Here's the log:
[latest.log](https://github.com/MehVahdJukaar/sleep_tight/files/11359294/latest.log)
[debug.log](https://github.com/MehVahdJukaar/sleep_tight/files/11359295/debug.log)

**Comment by MehVahdJukaar** [comment]
im checking that. be aware that if sleep warap does what i think it does most of the mod shouldnt work anyways as bedbugs, nightmares and mob encounters work on the assumption night is skipped instantaneously, allowing them to alter the time when you do wake up

**Comment by MehVahdJukaar** [comment]
yeah i just chekced with that mod on. I cant replicate any crash when hitting a mob but the mod itself invalidates almost the entirety of sleep tight as just for the way it works sleep cooldown, bed benefits, bedbugs, nightmares and bed encounters dont work since they apply after a successful night slept which there never triggers. Its just conceptually incompatible

**Comment by BLUEChester** [comment]
Ahh rip, I appreciate that you let me know though I hadn't even realized it was doing that.

It could be cool if Sleep Tight added being able to sleep with just 1 player in a server, as this is the main reason I have sleep warp installed, and is a must-have for me and my server. Totally understand if this isn't in the vision of the mod though.

-

Unfortunately, once again I'm experiencing an issue with Very Many Players & Sleep Tight, which is really weird as it seems to not happen sometimes, but be really consistent other times.
It crashes the game when punching/attacking any entity, in my test I broke an armor stand with Sleep Tight installed (w/o VMP installed), and then with VMP installed (w/o Sleep Tight installed), and both times it doesn't crash. But when they are both installed together the game crashes.

Here's my log:
[debug.log](https://github.com/MehVahdJukaar/sleep_tight/files/11359431/debug.log)
[latest.log](https://github.com/MehVahdJukaar/sleep_tight/files/11359432/latest.log)

Thanks for your help and fast replies.

**Comment by MehVahdJukaar** [comment]
there's no need for that. Thats a vanilla feature controlled by a gamerule which controls the percentage of players that need to sleep

**Comment by MehVahdJukaar** [comment]
as for the issue i dont see any mod mentioned there except for c2me and unionlib and seems to be something with a mod trying to geet an atatck damage attribute or other damage realted attribute when you hit something. idk why sleep tight would have anything to do here as it doesnt alter any of that

**Comment by BLUEChester** [comment]
Yeah its weird, I just assume it is related as when I remove either sleep tight or vmp the issue fixes itself. I'll try removing c2me and union lib. Thanks again.

Also didn't realize there was a game rule, that will be really helpful thanks for letting me know.

**Comment by SettingDust** [comment]
https://mclo.gs/QUVE4Ma

Facing in 1.20.1 fabric.
No any other mod
